packages/client/components/TeamPromptMeeting.tsx

```diff
@@ -22,23 +86,76 @@ const TeamPromptMeeting = (props: Props) => {
     graphql`
       fragment TeamPromptMeeting_meeting on TeamPromptMeeting {
         ...TeamPromptTopBar_meeting
+        ...useMeeting_meeting
+        phases {
+          stages {
+            ... on TeamPromptResponseStage {
+              teamMember {
+                id
+                preferredName
+                picture
+              }
+            }
+          }
+        }
       }
     `,
     meetingRef
   )
 
+  const maybeTabletPlus = useBreakpoint(Breakpoint.FUZZY_TABLET)
+  const teamMembers = useMemo(() => {
+    return meeting.phases
+      .flatMap((phase) => phase.stages)
+      .flatMap((stage) => stage.teamMember)
+      .map((teamMember) => {
+        if (!teamMember) {
+          return null
+        }
+
+        return {
+          ...teamMember,
+          key: teamMember.id
+        }
+      })
+      .filter(isNotNull)
+  }, [meeting.phases])
```

Comment:
-1 with the changes above, you can simplify this to:

```ts
 const responsePhase = meeting.phases.find((phase) => phase.__typename === 'TeamPromptResponsePhase')!
 const teamMembers responsePhase.stages.map((stage) => {...stage.teamMember, key: stage.teamMember.id}))
```

Reply:
typescript is a little dumb in that case and `responsePhase` can be another type as well.
see the screenshot below
![Screenshot 2022-04-19 at 15 35 16](https://user-images.githubusercontent.com/1017620/164016059-9b03c96e-c12f-40fd-abf0-d74f9c2ee4e3.png)

do we already have some generic utility to generate type predicates? we'd need something like this and I just want to make sure I haven't missed anything in the codebase 
https://stackoverflow.com/questions/59050071/narrow-return-type-of-find-from-discriminated-union-array

Reply:
ah good call, we have `GQLType`, which does the trick. I'll write a little commit to showcase it